From 8256be6437868d03149e25e68e0f46ce812e4b96 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 27 Feb 2015 17:06:51 -0500
Subject: [PATCH 175/315] session-worker: add FIXME

Document a hang that I need to deal with in the near term.
---
 daemon/gdm-session-worker.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/daemon/gdm-session-worker.c b/daemon/gdm-session-worker.c
index 671a95a4..8ecd425b 100644
--- a/daemon/gdm-session-worker.c
+++ b/daemon/gdm-session-worker.c
@@ -995,6 +995,16 @@ jump_to_vt (GdmSessionWorker  *worker,
 
                 fd = worker->priv->session_tty_fd;
 
+                /* FIXME: I don't think this is quite right.
+                 * If the current VT is KD_TEXT/VT_AUTO mode, setting
+                 * KD_GRAPHICS here can cause deadlocks.
+                 *
+                 * We probably need to set the current vt to VT_PROCESS
+                 * mode, (in addition to the new vt)  only if it's in
+                 * KD_TEXT mode.
+                 *
+                 * This is yucky, we really should get the display managers to use logind
+                 */
                 g_debug ("GdmSessionWorker: first setting graphics mode to prevent flicker");
                 if (ioctl (fd, KDSETMODE, KD_GRAPHICS) < 0) {
                         g_debug ("GdmSessionWorker: couldn't set graphics mode: %m");
-- 
2.15.1

